### PR TITLE
Fix a line clipping edge case

### PR DESCRIPTION
**Pull request by pozdnyakov**
Fix the edge case when a line string enters the clip region from outside and has a node exactly on the clip region border.
Before, the node on the border was skipped, which caused wrong line metrics estimation.

### CODE REVIEW

File: include/mapbox/geojsonvt/clip.hpp [hunk 1]
```diff
@@ -186,7 +189,7 @@ class clipper {
         }
 
         if (!slice.empty()) { // add the final slice
-            slice.segEnd = lineLen;
+            if (lineMetrics) slice.segEnd = lineLen;
```
Comment (alexshalamov) [inline include/mapbox/geojsonvt/clip.hpp:192]: 🤔 that was a bug, right? We were not supposed to update `slice.segEnd` if line metrics is disabled.

**Review by alexshalamov — APPROVED** [review]
lgtm

File: include/mapbox/geojsonvt/clip.hpp (hunk 1 above)
Comment (pozdnyakov) [reply include/mapbox/geojsonvt/clip.hpp:192]: yes

File: include/mapbox/geojsonvt/clip.hpp [hunk 2]
```diff
@@ -115,6 +115,7 @@ class clipper {
             const auto& b = line[i + 1];
             const double ak = get<I>(a);
             const double bk = get<I>(b);
+            const bool lastSegFlag = (i == (len - 2));
```
Comment (mourner) [inline include/mapbox/geojsonvt/clip.hpp:118]: Nit: I don't like the word "flag" in variables, maybe `isLastSeg` would sound nicer?

File: include/mapbox/geojsonvt/clip.hpp [hunk 3]
```diff
@@ -135,9 +136,10 @@ class clipper {
                     t = calc_progress<I>(a, b, k1);
                     slice.emplace_back(intersect<I>(a, b, k1, t));
                     if (lineMetrics) slice.segStart = lineLen + segLen * t;
-
-                    if (i == len - 2)
-                        slice.emplace_back(b); // last point
+                    if (lastSegFlag) slice.emplace_back(b); // last point
+                } else if (bk == k1 && !lastSegFlag) { // --->|..  |
```
Comment (mourner) [inline include/mapbox/geojsonvt/clip.hpp:140]: Nit: add an empty line above so that all the cases are visually separated.

**Review by mourner — APPROVED** [review]

**Comment by mourner** [comment]
@pozdnyakov FYI when porting this to JS, it turned out that this 2-character change fixes the original issue https://github.com/mapbox/geojson-vt/pull/161, so we can revert this somewhat convoluted fix with a much simpler change (that might fix other things).

**Pull request merged**